### PR TITLE
[Autocomplete] Reopen popover if popoverWasActive and there are options

### PR DESCRIPTION
**Pull request by solonaarmstrong**
### WHY are these changes introduced?

Resolves https://github.com/Shopify/polaris-react/issues/467

### WHAT is this pull request doing?

Adds `popoverWasActive` state and sets popover to active if there are options again. I'm not sure this is testable.

popoverwasactive

## 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
/* eslint-disable */

import * as React from 'react';
import {Page, AppProvider, Stack, Tag, TextContainer} from '@shopify/polaris';
import Autocomplete from '../src/components/Autocomplete';
import _ from 'lodash';

interface Props {}
interface State {}

export default class DemoTagsAutocomplete extends React.Component<
  Props,
  State
> {
  // Quick Config
  ALLOW_MULTIPLE = true;
  RESULT_PAGE_LENGTH = 10;

  options = [
    {value: 'byward_market', label: 'Byward Market'},
    {value: 'cheese_pizza', label: 'Cheese Pizza'},
    {value: 'macaroni_pizza', label: 'Macaroni Pizza'},
    {value: 'pepperoni_pizza', label: 'Pepperoni Pizza'},
    {value: 'centretown', label: 'Centretown'},
    {value: 'hintonburg', label: 'Hintonburg'},
    {value: 'pineapple', label: 'Pineapple'},
    {value: 'pineapple_pizza', label: 'Pineapple Pizza'},
    {value: 'chinatown', label: 'Chinatown'},
    {value: 'kraft_singles', label: 'Kraft Singles'},
    {value: 'dasani_water', label: 'Dasani Water'},
    {value: 'ice_soup', label: 'Ice Soup'},
    {value: 'chana_masala', label: 'Chana Masala'},
    {value: 'eggs', label: 'Eggs'},
    {value: 'mango', label: 'Mango'},
    {value: 'butter_chicken', label: 'Butter Chicken'},
    {value: 'deprived_snakes', label: 'Deprived Snakes'},
    {value: 'pizza_pockets', label: 'Pizza Pockets'},
  ];

  state = {
    selected: [],
    inputText: '',
    options: this.options.slice(0, this.RESULT_PAGE_LENGTH),
    filteredOptions: this.options.slice(
      this.RESULT_PAGE_LENGTH,
      this.options.length - 1,
    ),
    loading: false,
  };

  updateText = (newValue: string) => {
    this.setState({inputText: newValue});
    this.filterAndUpdateOptions(newValue);
  };

  removeTag = (tag: string) => {
    let newSelected: string[] = this.state.selected;
    newSelected.splice(newSelected.indexOf(tag), 1);
    this.setState({selected: newSelected});
  };

  addTag = (tag: string) => {
    if (!tag || tag === '') {
      return;
    }
    let tagLabel = '';
    tagLabel = tag.replace('_', ' ');
    tagLabel = _.startCase(tagLabel);

    this.options.push({value: tag, label: tagLabel});
    this.filterAndUpdateOptions(tag);
  };

  renderTags = (): React.ReactNode => {
    return this.state.selected.map((option: string) => {
      let tagLabel = '';
      tagLabel = option.replace('_', ' ');
      tagLabel = _.startCase(tagLabel);
      return (
        <Tag key={'option' + option} onRemove={() => this.removeTag(option)}>
          {tagLabel}
        </Tag>
      );
    });
  };

  shouldLoadMoreResults = (): boolean => {
    const {filteredOptions, loading} = this.state;
    if (filteredOptions.length === 0 || loading) {
      return false;
    }
    this.setState({loading: true});
    return true;
  };

  loadMoreResults = () => {
    const {filteredOptions, options} = this.state;
    if (this.shouldLoadMoreResults()) {
      let currentlyLoadedOptions = options;
      let unloadedOptions = filteredOptions;

      const endIndex =
        currentlyLoadedOptions.length + this.RESULT_PAGE_LENGTH - 1;
      let optionsToLoad = unloadedOptions.slice(0, endIndex);

      currentlyLoadedOptions = currentlyLoadedOptions.concat(optionsToLoad);
      unloadedOptions = unloadedOptions.slice(
        endIndex,
        unloadedOptions.length - 1,
      );

      setTimeout(() => {
        this.setState({
          options: currentlyLoadedOptions,
          filteredOptions: unloadedOptions,
          loading: false,
        });
      }, 300);
    }
  };

  shouldShowRenderActionButton = (): boolean => {
    const {inputText} = this.state;
    return !(
      !inputText ||
      inputText === '' ||
      (this.ALLOW_MULTIPLE && containsOption(this.options, inputText))
    );
  };

  getListTitle = (): string | undefined => {
    if (this.shouldShowRenderActionButton()) {
      return undefined;
    }
    return 'Frequently Used Tags';
  };

  render() {
    return (
      <AppProvider>
        <Page title="Demo Tags Autocomplete">
          <br />
          <TextContainer>
            <Stack>{this.renderTags()}</Stack>
          </TextContainer>
          <br />
          <Autocomplete
            options={this.state.options}
            selected={this.state.selected}
            textField={
              <Autocomplete.ComboBox.TextField
                onChange={this.updateText}
                label=""
                value={this.state.inputText}
                placeholder="Vintage, cotton, summer"
              />
            }
            onSelect={this.updateSelection}
            allowMultiple={this.ALLOW_MULTIPLE}
            loading={this.state.loading}
            willLoadMoreResults
            onLoadMoreResults={this.loadMoreResults}
            listTitle={this.getListTitle()}
          />
        </Page>
      </AppProvider>
    );
  }

  filterAndUpdateOptions = (inputString: string) => {
    if (inputString === '') {
      this.setState({
        options: this.options.slice(0, this.RESULT_PAGE_LENGTH),
        filteredOptions: this.options.slice(
          this.RESULT_PAGE_LENGTH,
          this.options.length,
        ),
      });
      return;
    }

    const filterRegex = new RegExp(inputString, 'i');
    const resultOptions = this.options.filter((option) =>
      option.label.match(filterRegex),
    );
    let endIndex = resultOptions.length - 1;
    if (resultOptions.length === 0) {
      endIndex = 0;
    }
    this.setState({
      options: resultOptions.slice(0, this.RESULT_PAGE_LENGTH),
      filteredOptions: resultOptions.slice(this.RESULT_PAGE_LENGTH, endIndex),
    });
  };

  updateSelection = (updatedSelection: string[]) => {
    const selectedText = updatedSelection.map((selectedItem: string) => {
      const matchedOption = this.options.filter((option) => {
        return option.value.match(selectedItem);
      });
      return matchedOption[0] && matchedOption[0].label;
    });
    if (this.ALLOW_MULTIPLE) {
      this.setState({selected: updatedSelection});
    } else {
      this.setState({selected: selectedText, inputText: selectedText});
    }
  };
}

function containsOption(a: any, optionString: string) {
  for (var i = 0; i < a.length; i++) {
    const filterRegex = new RegExp(
      '^' + optionString.replace(' ', '_') + '$',
      'i',
    );
    if (a[i].value.match(filterRegex)) {
      return true;
    }
  }
  return false;
}

/* eslint-enable */

```

</details>

### 🎩 checklist

* [ ] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [ ] Updated [UNRELEASED.md](https://github.com/Shopify/polaris-react/blob/master/UNRELEASED.md)

### CODE REVIEW

File: src/components/index.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ export {
 
 export {
   default as Autocomplete,
-  Props as AutcompleteProps,
+  Props as AutocompleteProps,
```
Comment (solonaarmstrong) [inline src/components/index.ts]: Fixing a typo

**Comment by darrenhebner** [comment]
Maybe not exactly related to your changes here, but I just want to leave a note before we add more logic. It's getting pretty tricky to track the popover state in the combobox.

For example, every time you click on an option in the popover `handleBlur`, `handleFocus`, and `handleClick` are all triggered. `handleFocus` and `handleClick` set `popoverActive` to true, while `handleBlur` sets it to false.

File: src/components/Autocomplete/components/ComboBox/ComboBox.tsx [hunk 2]
```diff
@@ -177,6 +179,13 @@ export default class ComboBox extends React.PureComponent<Props, State> {
     ) {
       // eslint-disable-next-line react/no-did-update-set-state
       this.setState({popoverActive: false});
+    } else if (
+      popoverWasActive &&
+      navigableOptions &&
+      navigableOptions.length !== 0
```
Comment (tmlayton) [inline src/components/Autocomplete/components/ComboBox/ComboBox.tsx]:
```suggestion
      navigableOptions.length > 0
```

**Comment by solonaarmstrong** [comment]
@tmlayton In the original PR description, there's a gif that shows how the popover closes if there are no results (because the popover no longer contains content). We need a way to tell the popover to open again if the search text changes and there are now results to show. If there's a better solution to what I implemented, I'm happy to change it to something more intelligent.

**Comment by solonaarmstrong** [comment]
Thank you, @tmlayton

**Pull request merged**